packages/metals-vscode/src/extension.ts

```diff
@@ -905,7 +916,23 @@ function launchMetals(
       });
 
       registerCommand("metals.toggle-show-inferred-type", () => {
-        toggleBooleanWorkspaceSetting("showInferredType");
+        const setting = "showInferredType";
+        const config = workspace.getConfiguration("metals");
+        const configProperty = config.inspect<string>(setting);
+        const currentValue = configProperty?.workspaceValue ?? "false";
+        let newValue = "true";
+        switch (currentValue) {
+          case "true":
+            newValue = "minimal";
+            break;
+          case "minimal":
+            newValue = "false";
+            break;
+          case "false":
+            newValue = "true";
+            break;
+        }
```

Comment:
Honestly, I'm a bit lost with this logic. 
Why `false` turns into `true` and `minimal` into `false`? 😸 
 
Do I understand it correctly:
- before launching server we do boolean to string migration in `migrateOldSetting`
- then we register a commend and perform some additional migration  into enum?

Reply:
> then we register a command and perform some additional migration into enum?

That's only for toggle. Previously it was only `true-> false` and `false -> true`, but now we have 3 possible options and it's still useful to have a quick option to switch things on and off. So I decided on `false -> true -> minimal -> false` order.

Reply:
@tgodzik oh - got it, thanks for explanation!

Reply:
> false -> true -> minimal -> false

That makes sense 👍